common/src/test/java/org/apache/hertzbeat/common/queue/impl/RedisCommonDataQueueTest.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hertzbeat.common.queue.impl;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.hubspot.jackson.datatype.protobuf.ProtobufModule;
+import io.lettuce.core.RedisClient;
+import io.lettuce.core.RedisURI;
+import io.lettuce.core.api.StatefulRedisConnection;
+import io.lettuce.core.api.sync.RedisCommands;
+import org.apache.hertzbeat.common.config.CommonProperties;
+import org.apache.hertzbeat.common.entity.alerter.Alert;
+import org.apache.hertzbeat.common.entity.message.CollectRep;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.utility.DockerImageName;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * test for {@link RedisCommonDataQueue}
+ */
+class RedisCommonDataQueueTest {
+
+	private static GenericContainer<?> redisContainer;
+	private RedisCommonDataQueue redisCommonDataQueue;
+	private RedisClient redisClient;
+	private StatefulRedisConnection<String, String> connection;
+	private RedisCommands<String, String> syncCommands;
+	private ObjectMapper objectMapper;
+
+	@BeforeAll
+	public static void startContainer() {
+
+		redisContainer = new GenericContainer<>(
+				DockerImageName.parse("redis:latest"))
+				.withExposedPorts(6379);
```

Comment:
hi, it seem will create a docker container by unit test code, not recommended to use this in `unit test cases`. Maybe in `integration tests` we can create container by config instead of the code.

Reply:
yep. got it.